### PR TITLE
update-design-doc-with-explicit-sub-scope-changes

### LINKED ISSUE #163: Subscription-Issue4: Need to change the scope pattern for explicit subscriptions
**Problem description**
The API design guidelines state that scopes should be represented as:

- API Name: address-management, numbering-information...
- Protected Resource: orders, billings…
- Grant-level, action on resource: read, write…

For explicit subscriptions, giving an example of device-status, this pattern results in 

`device-status:subscriptions:create`

And with this subscription, any one of the below event types could be subscribed to:

```
org.camaraproject.device-status.v0.roaming-status,
org.camaraproject.device-status.v0.roaming-on,
org.camaraproject.device-status.v0.roaming-off,
org.camaraproject.device-status.v0.roaming-change-country,
org.camaraproject.device-status.v0.connectivity-data,
org.camaraproject.device-status.v0.connectivity-sms,
org.camaraproject.device-status.v0.connectivity-disconnected,
```
The scope device-status:subscriptions:create doesn't give any insights on the event types.

**Expected behavior**

- For explicit subscriptions, the scope should give some insights into which event types could be subscribed to.

For e.g.
`device:roaming-status-sub:create`

- The new pattern should be documented in API design guidelines for explicit subscriptions
- The subprojects who have implemented this pattern for explicit subscriptions should make changes as agreed in design guidelines.

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by shilpa-padgaonkar**
#### What type of PR is this?

Add one of the following kinds:
* correction

#### What this PR does / why we need it:
Explicit subscriptions need to be transparent about the event types that they offer.
Following changes are included:
- Suggests that explicit subscriptions should have a separate API file
- API name must include the keyword "subscriptions" at the end
- Suggest how scope should be represented when actions are create (subscription) and when they are delete and read (subscription)

#### Which issue(s) this PR fixes:

Fixes #163 

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by shilpa-padgaonkar** [comment]
Changed PR to draft based on subscription related discussion. Also need to await feedback & consensus on https://github.com/camaraproject/Commonalities/issues/184

**Comment by shilpa-padgaonkar** [comment]
@bigludo7 and @PedroDiez : Would be great if you could kindly review the scope related changes for explicit subscriptions documented in the design guidelines and recommend changes if needed.

File: documentation/API-design-guidelines.md
Comment (bigludo7) [inline documentation/API-design-guidelines.md:1357]: keyword - not keyyword

File: documentation/API-design-guidelines.md
Comment (bigludo7) [inline documentation/API-design-guidelines.md:976]: :create or :read ? I'm fine with either

File: documentation/API-design-guidelines.md
Comment (bigludo7) [inline documentation/API-design-guidelines.md:970]: Should we explicitly write that we are defining here a scope for each event-type in order to manage subscribed consent at this level for subscription creation?

**Review by bigludo7 — APPROVED** [review]
Look good for me @shilpa-padgaonkar . I've just 3 comments for your appreciation (not blocking) and one word to correct. Thanks.

File: documentation/API-design-guidelines.md
Comment (shilpa-padgaonkar) [inline documentation/API-design-guidelines.md:969]:
```suggestion
The format to define scopes for explicit subscriptions with action create, includes the event type in its formulation to ensure that consent is managed at the level of subscribed event types. Scopes should be represented as below for APIs that offer explicit event subscriptions with action create:
```

File: documentation/API-design-guidelines.md
Comment (shilpa-padgaonkar) [reply documentation/API-design-guidelines.md:970]: done

File: documentation/API-design-guidelines.md
Comment (shilpa-padgaonkar) [inline documentation/API-design-guidelines.md:965]:
```suggestion
- Grant-level, action on resource: read, delete
This type of formulation is not needed for the create action.
```

File: documentation/API-design-guidelines.md
Comment (shilpa-padgaonkar) [reply documentation/API-design-guidelines.md:967]: done

File: documentation/API-design-guidelines.md
Comment (shilpa-padgaonkar) [inline documentation/API-design-guidelines.md:1358]:
```suggestion
To ensure consistency across Camara subprojects, it is necessary that explicit subscriptions are handled within separate API/s. It is recommended when possible to append the keyword "subscriptions" at the end of the API name. For e.g. device-roaming-subscriptions.yaml
```

File: documentation/API-design-guidelines.md
Comment (shilpa-padgaonkar) [reply documentation/API-design-guidelines.md]: done

**Review by bigludo7 — APPROVED** [review]

File: documentation/API-design-guidelines.md
Comment (PedroDiez) [reply documentation/API-design-guidelines.md:975]: :create is fine

File: documentation/API-design-guidelines.md
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1356]:
```suggestion
To ensure consistency across Camara subprojects, it is necessary that explicit subscriptions are handled within separate API/s. It is mandatory to append the keyword "subscriptions" at the end of the API name. For e.g. device-roaming-subscriptions.yaml
```

**Review by PedroDiez — DISMISSED** [review]
Minor format point

LGTM in advance

**Comment by shilpa-padgaonkar** [comment]
@PedroDiez and @bigludo7 have already approved the PR, but I would need a codeowner approval to be able to merge this.  @rartych is on vacation. Could either @patrice-conil or @RubenBG7 kindly approve the PR?

**Comment by bigludo7** [comment]
@shilpa-padgaonkar my colleague Patrice is also enjoying well-deserved vacation :)
Hope @RubenBG7 will be able to help.

**Review by RubenBG7 — APPROVED** [review]

**Comment by RubenBG7** [comment]
Done @shilpa-padgaonkar

**Pull request merged**
